lib_gb/src/machine/gameboy.rs

```diff
@@ -88,16 +90,23 @@ impl<'a> GameBoy<'a>{
             }
             
             //PPU
-            self.cycles_counter += cpu_cycles_passed as u32 + interrupt_cycles as u32;
-            self.ppu.update_gb_screen(&self.mmu, self.cycles_counter);
+            let iter_total_cycles= cpu_cycles_passed as u32 + interrupt_cycles as u32;
+            self.ppu.update_gb_screen(&self.mmu, iter_total_cycles);
             //updating after the PPU
             self.register_handler.update_registers_state(&mut self.mmu, &mut self.cpu, &mut self.ppu, &mut self.interrupts_handler, &joypad, 0);
+
+            if last_ppu_power_state != self.ppu.screen_enable && self.ppu.screen_enable{
```

Comment:
Could be done more elegantly.
If the last power state is off and the current one is on